src/util/antoraSupport.ts

```diff
@@ -0,0 +1,39 @@
+import vscode from 'vscode'
+import fs from 'fs'
+import yaml from 'js-yaml'
+import path from 'path'
+
+export default class AntoraSupport {
+  public async isDetected (): Promise<boolean> {
+    const antoraYmlFiles = await vscode.workspace.findFiles('**/antora.yml', '/node_modules/', 10)
+    return antoraYmlFiles.length > 0
+  }
+
+  public async getAntoraYmlPath (textDocument: vscode.TextDocument): Promise<any> {
+    const pathToAsciidocFile = textDocument.uri.fsPath
+    const antoraYmlFiles = await vscode.workspace.findFiles('**/antora.yml', '/node_modules/', 100)
+    // check if antora.yml file matches with textDocument content source root
+    for (const antoraYmlFile of antoraYmlFiles) {
+      const asciidocFilename = path.basename(pathToAsciidocFile)
+      const modulesPath = path.join(path.dirname(antoraYmlFile.path), 'modules')
+      if (pathToAsciidocFile.startsWith(modulesPath) && pathToAsciidocFile.slice(modulesPath.length).match(/^\/[^/]+\/pages\/.*/) && asciidocFilename !== 'README.adoc') {
+        return antoraYmlFile.path
+      }
+      console.log('Antora file not applicable to current asciidoc document.')
+    }
+  }
+
+  public async getAttributes (textDocument: vscode.TextDocument): Promise<{ [key: string]: string }> {
+    try {
+      const doc = yaml.load(fs.readFileSync(await this.getAntoraYmlPath(textDocument), 'utf8'))
+      return doc.asciidoc.attributes
+    } catch (err) {
+      console.log(`Unable to parse ${await this.getAntoraYmlPath(textDocument)}, cause:` + err.toString())
```

Comment:
Since we call `await this.getAntoraYmlPath(textDocument)` again we will do the process twice. We should extract the result of the method call:

```js
const antoraYmlPath = await this.getAntoraYmlPath(textDocument)
```